### ISSUE TITLE
Blinking favourite record

### ISSUE THREAD

**Issue by BOHEUS** [twentyhq/twenty#9299]
**Scenario:**
1. Log in
2. Go to any record's details tab
3. Click favourite button

**Actual:** Newly favourited record appears, disappears and appears again in less than 1 second

https://github.com/user-attachments/assets/e1429d69-5b2e-484d-ab76-865cb9b434e7

**Expected:** Newly favourited record should appear and don't blink

**Comment by ehconitin** [comment]
yes I can, looking into it

**Comment by ehconitin** [comment]
@FelixMalfait The recent change to `cache-and-network` fetch policy is causing a UI flicker in favorites. 
https://github.com/twentyhq/twenty/pull/9296

When creating a favorite:
1. First optimistic update shows the data
2. Cache gets temporarily cleared during network request (`[]`)
3. Second optimistic update restores the data

I think this is happening because `cache-and-network` makes a network request even with cached data? -- not so sure about this 

Would you prefer to:
1. Handle this at the cache level (modify optimistic updates)
2. Handle this at the UI level (preserve the previous state during updates)

**Comment by FelixMalfait** [comment]
@ehconitin good catch. I'm afraid it introduce the same flickering on many parts of the app.

Here's a suggestion on Stackoverflow: https://stackoverflow.com/questions/54279306/flickering-with-apollo-queries

Are you sure that data is really reset to [] or it's just the load state which is set to true? If the former (as suggested in the Stackoverflow response), it should be easier

In any case, this might require a change across many files :(. Could you give it a try in a few places where it looks doable? (e.g. look for `SkeletonLoader`). Thanks!

cc @charlesBochet @lucasbordeau fyi

**Comment by lucasbordeau** [comment]
Two ideas : 
- For now change the fetch policy back to previous for the problematic requests only
- In depth review of our home made optimistic rendering layer which might not be adapted anymore to the new fetch policy.

**Comment by FelixMalfait** [comment]
I think the optimistic rendering layer is fine, there's no need to change it.
The issue imo is around loading interfaces, as pointed in the Stackoverflow answer, we should do {if loading = true and data is null}, rather than just if {loading = true}

**Comment by lucasbordeau** [comment]
The difficulty is that we might want to allow null or [] data no ? How do we know for sure that when loading is false the returned data should be considered as "still loading" ?

**Comment by FelixMalfait** [comment]
Yes I think it's fine to show a loading state while we re-fetch network if the cache was empty (even if it was empty after a first fetch). It's actually nice for an empty state to have a way for users to 'refetch' and make sure that there was actually a refetch
